examples/StreamLayerReadExample.cpp

```diff
@@ -0,0 +1,231 @@
+  auto task_scheduler =
+      olp::client::OlpClientSettingsFactory::CreateDefaultTaskScheduler();
+  // Create a network client
+  auto http_client = olp::client::OlpClientSettingsFactory::
+      CreateDefaultNetworkRequestHandler();
+
+  // Initialize authentication settings
+  olp::authentication::Settings settings({access_key.id, access_key.secret});
+  settings.task_scheduler = std::move(task_scheduler);
+  settings.network_request_handler = http_client;
+  // Setup AuthenticationSettings with a default token provider that will
+  // retrieve an OAuth 2.0 token from OLP.
+  olp::client::AuthenticationSettings auth_settings;
+  auth_settings.provider =
+      olp::authentication::TokenProviderDefault(std::move(settings));
+
+  // Setup OlpClientSettings and provide it to the StreamLayerClient.
+  olp::client::OlpClientSettings client_settings;
+  client_settings.authentication_settings = auth_settings;
+  client_settings.network_request_handler = std::move(http_client);
+
+  // Create subscription, used kSerial or kParallel subscription mode
+  olp::dataservice::read::SubscribeRequest subscribe_request;
+  subscribe_request.WithSubscriptionMode(subscription_mode);
+
+  // set options, default values described here:
+  // https://developer.here.com/olp/documentation/data-api/api-reference-stream.html
+  olp::dataservice::read::ConsumerOptions expected_options = {
+      olp::dataservice::read::ConsumerOption("auto.commit.interval.ms",
+                                             int32_t{5000}),
+      olp::dataservice::read::ConsumerOption("auto.offset.reset", "earliest"),
+      olp::dataservice::read::ConsumerOption("enable.auto.commit", "false"),
+      olp::dataservice::read::ConsumerOption("fetch.max.bytes",
+                                             int32_t{52428800}),
+      olp::dataservice::read::ConsumerOption("fetch.max.wait.ms", int32_t{500}),
+      olp::dataservice::read::ConsumerOption("fetch.min.bytes", int32_t{1}),
+      olp::dataservice::read::ConsumerOption("group.id", "group_id_1"),
+      olp::dataservice::read::ConsumerOption("max.partition.fetch.bytes",
+                                             int32_t{1048576}),
+      olp::dataservice::read::ConsumerOption("max.poll.records", int32_t{500})};
+  subscribe_request.WithConsumerProperties(
+      olp::dataservice::read::ConsumerProperties(expected_options));
+
+  auto read_from_stream_layer = [&]() {
+    // Create stream layer client with settings and catalog, layer specified
+    olp::dataservice::read::StreamLayerClient client(olp::client::HRN{catalog},
+                                                     layer_id, client_settings);
+    if (!CreateSubscription(client, subscribe_request)) {
+      return false;
+    }
+    RunPoll(client);
+    if (!DeleteSubscription(client)) {
+      return false;
+    }
+    return true;
+  };
+
+  int result = 0;
+  if (subscription_mode ==
+      olp::dataservice::read::SubscribeRequest::SubscriptionMode::kParallel) {
```

Comment:
For the sake of readability maybe add an alias for the namespace olp::dataservice::read or for the SubscribeRequest.